Userland/Libraries/LibGUI/AbstractView.cpp

```diff
@@ -717,31 +717,30 @@ void AbstractView::draw_item_text(Gfx::Painter& painter, ModelIndex const& index
         text_color = is_focused() ? palette().selection_text() : palette().inactive_selection_text();
     else
         text_color = index.data(ModelRole::ForegroundColor).to_color(palette().color(foreground_role()));
+
     if (index == m_highlighted_search_index) {
-        Utf8View searching_text(m_highlighted_search);
-        auto searching_length = searching_text.length();
-        if (searching_length > search_highlighting_offset)
-            searching_length -= search_highlighting_offset;
-        else if (search_highlighting_offset > 0)
-            searching_length = 0;
+        Utf8View searching_text(m_highlighted_search.substring_view(min(search_highlighting_offset, m_highlighted_search.length())));
 
         // Highlight the text background first
-        auto background_searching_length = searching_length;
-        painter.draw_text([&](Gfx::FloatRect const& rect, Utf8CodePointIterator&) {
-            if (background_searching_length > 0) {
-                background_searching_length--;
-                painter.fill_rect(rect.to_type<int>().inflated(0, 2), palette().highlight_searching());
-            }
-        },
-            text_rect, item_text, font, alignment, elision);
+        auto highlight_rect = text_rect.shrunken(0, text_rect.height() - font.glyph_height() - 2);
+        highlight_rect.set_width(min(font.width(searching_text), font.width(item_text)));
+        if (alignment == Gfx::TextAlignment::Center) {
+            // If the text is center aligned, the highlight rect needs to be shifted to the right so that the two line up
+            highlight_rect.translate_by((text_rect.width() - font.width(item_text)) / 2, 0);
```

Comment:
Above, you use the minimum of the searched text and the item text for the highlight width. But here, you just use the item text width. I think that can lead to some weird highlight positions on centered text. I'd suggest extracting that min() call above into a variable (`highlight_width` or similar) and reusing the variable here in the translate_by calculation.

Reply:
Thanks for the feedback!

What is being calculated with the `(text_rect.width() - font.width(item_text)) / 2` expression is the distance between the leftmost edge of the text rect and the start of the item text. In the example screenshot above, that would be the 3 pixels between the start of the dotted line and the f in "foo".

The reason I don't use `highlight_width` or similar in this calculation is because that would simply center the highlight rect, which would not be correct in cases where the highlight rect is narrower than the item text.

Reply:
Okay, makes sense, you can resolve this comment then!